src/app/functions/server/getRelevantGuidedSteps.ts

```diff
@@ -187,22 +179,20 @@ function isEligibleFor(data: InputData, stepId: StepLink["id"]): boolean {
 }
 
 function hasCompleted(data: InputData, stepId: StepLink["id"]): boolean {
-  if (stepId === "StartScan") {
-    return data.latestScanData?.scan !== null;
-  }
-
-  if (stepId === "ScanResult") {
-    return (
-      data.latestScanData !== null &&
-      Array.isArray(data.latestScanData?.results) &&
-      !data.latestScanData.results.some((scanResult) => {
-        return scanResult.status === "new" && !scanResult.manually_resolved;
-      }) &&
-      data.latestScanData?.scan?.onerep_scan_status === "finished"
-    );
+  if (stepId === "Scan") {
+    const hasRunScan =
+      typeof data.latestScanData?.scan === "object" &&
+      data.latestScanData?.scan !== null;
+    const hasResolvedAllScanResults =
+      data.latestScanData?.scan?.onerep_scan_status === "finished" &&
```

Comment:
I think we also landed on counting scan results that have a status of `in_progress` as “resolved”. I guess that will only be relevant for automatic removal, though.

Reply:
Yeah I think you're right; once your permutations work lands we can add proper tests that actually verify the right links show up in the right situations (which you already mentioned in your test PR), which should've caught this. Nice :eagle: :eye: ! Fixed in 6f5ee8ba2.

Reply:
Unit test is failing, but that is because the "scan in progress" notice only gets shown if the current step is `"Scan"`, which it no longer is with this change. I believe the PR you just merged changes that, so after a rebase and fixing the conflicts, it should succeed again.